### DISCUSSION TITLE
Is the agent suppose to keep running when hub disconnects?

### DISCUSSION THREAD

**Discussion by crosspolar** [henrygd/beszel#258] [Support] [answered]
I have the agent running on my server and the hub running on my laptop. When I shutdown my laptop in the evening, I also disconnect of course hub and agent. However, when I start the Hub again in the morning it looks like agent just started because all the graphs are empty before.
![image](https://github.com/user-attachments/assets/64945290-0c45-4dd2-ac24-e8bd360f71be)

Is it suppose to be like that or am I doing something wrong?

**Comment by henrygd** [comment, marked as answer]
The hub should be on the server and be running at all times. It's where the data for all machines (agents) is collected and stored in the database.

The agent doesn't do anything unless the hub requests information from it. If the hub is shut down, the agent will do absolutely nothing until the hub comes back online to request data from it. The agent does not have its own database to store backlogged data.
